add-chain/e2e_test.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/urfave/cli/v2"
+	"gopkg.in/yaml.v2"
+)
+
+func TestCLIApp(t *testing.T) {
+	app := &cli.App{
+		Name:   "add-chain",
+		Usage:  "Add a new chain to the superchain-registry",
+		Flags:  []cli.Flag{ChainTypeFlag, TestFlag},
+		Action: entrypoint,
+	}
+
+	args := []string{"add-chain", "-chain-type", "standard", "-test", "true"}
+	err := app.Run(args)
+	if err != nil {
+		t.Errorf("add-chain app failed: %v", err)
+	}
+
+	yamlEqual, err := checkConfigYaml()
+	if err != nil {
+		t.Errorf("failed to read yaml config files: %v", err)
+	}
+	if !yamlEqual {
+		t.Error("test config yaml file does not match expected file")
+	}
+
+	jsonEqual, err := compareJsonFiles("./testdata/superchain/extra/addresses/sepolia/")
+	if err != nil {
+		t.Errorf("failed to read json address files: %v", err)
+	}
+	if !jsonEqual {
+		t.Error("test json address file does not match expected file")
+	}
+
+	jsonEqual, err = compareJsonFiles("./testdata/superchain/extra/genesis-system-configs/sepolia/")
+	if err != nil {
+		t.Errorf("failed to read json genesis files: %v", err)
+	}
+	if !jsonEqual {
+		t.Error("test json genesis file does not match expected file")
+	}
+}
+
+func compareJsonFiles(dirPath string) (bool, error) {
+	expectedBytes, err := os.ReadFile(dirPath + "expected.json")
+	if err != nil {
+		return false, err
+	}
+
+	var expectJSON map[string]interface{}
+	if err := json.Unmarshal(expectedBytes, &expectJSON); err != nil {
+		return false, err
+	}
+
+	testBytes, err := os.ReadFile(dirPath + "awesomechain.json")
+	if err != nil {
+		return false, err
+	}
+
+	var testJSON map[string]interface{}
+	if err := json.Unmarshal(testBytes, &testJSON); err != nil {
+		return false, err
+	}
+
+	return reflect.DeepEqual(expectJSON, testJSON), nil
+}
+
+func checkConfigYaml() (bool, error) {
+	expectedBytes, err := os.ReadFile("./testdata/superchain/configs/sepolia/expected.yaml")
+	if err != nil {
+		return false, err
+	}
+
+	var expectYaml RollupConfig
+	if err := yaml.Unmarshal(expectedBytes, &expectYaml); err != nil {
+		return false, err
+	}
+
+	testBytes, err := os.ReadFile("./testdata/superchain/configs/sepolia/awesomechain.yaml")
+	if err != nil {
+		return false, err
+	}
+
+	var testYaml RollupConfig
+	if err := yaml.Unmarshal(testBytes, &testYaml); err != nil {
+		return false, err
+	}
+
+	return reflect.DeepEqual(expectYaml, testYaml), nil
```

Comment:
Will this check fail on whitespace changes? Maybe a nit, but I think it's nice for us to be strict about the overal format of the document. Could we just compare the raw bytes of the file?

Reply:
Updated in https://github.com/ethereum-optimism/superchain-registry/pull/204/commits/bcdfb7cb102f0c8e752794ff7b51a3af081154a2